### DISCUSSION TITLE
migrating from opentracing to opentelemetry

### DISCUSSION THREAD

**Discussion by GregJohnStewart** [quarkusio/quarkus#31557] [Q&A] [answered]
So looking at migrating from opentracing to opentelemetry and the process is less than transparent.

 - Noticing the loss of @Traced, does this mean everything is traced now?
 - How long will openTracing be supported for? Trying to get the best guidance for a client, and hard to recommend if deprecated. Looks like the version of Distributed Tracing on OCP is still in-dev/preview, also making it hard to recommend at this time
 - Does one still need the micrometer-registry-prometheus extension? Does Otel cover that, or they a separate item?

[a comment by an automated account is not shown]

**Comment by brunobat** [comment, marked as answer]
Hi @GregJohnStewart 
1) `@Traced`is Replaced by `@WithSpan`
2) OpenTracing doesn't have a deprecation date but we incentivise to migrate to OpenTelemetry because OpenTracing is in end-of-life.
3) You can use a micrometer-registry-otlp from quarkiverse: https://github.com/quarkiverse/quarkus-micrometer-registry/tree/main/micrometer-registry-otlp

Also, you can check this project for guidance in Migrating from OpenTracing to OpenTelemetry: https://github.com/brunobat/quarkus-observability-demo/tree/main/quarkus-observability-demo-opentracing-shim 
... but is too thin. I'm writing a blog on how to help with migrations

**Reply by GregJohnStewart** [reply]
Thanks! This does clear some things up, though a little surprised there isn't a comprehensive migration guide

**Reply by brunobat** [reply]
You can check this presentation I did at JakartaOne. I explain how to migrate things there... https://www.youtube.com/watch?v=Xzm9m0PG5Wk 
I hope I was not too clunky :)

**Reply by GregJohnStewart** [reply]
Thanks! That video is super helpful!
Just looks to be a tedious task (migrating) more than anything; where `Traced` was able to be class-level and now `WithSpan` is only for methods

**Reply by brunobat** [reply]
Can I mark the question as answered?

**Reply by GregJohnStewart** [reply]
I think so! Thanks again!
